geode-concurrency-test/src/main/java/org/apache/geode/test/concurrency/ParallelExecutor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.test.concurrency;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * Executor that executes multiple tasks in parallel as part of the main body of concurrent test.
+ *
+ * See {@link ConcurrentTestRunner}
+ */
+public interface ParallelExecutor {
+
+  /**
+   * Add a task to run in parallel
+   */
+  <T> Future<T> inParallel(Callable<T> callable);
+
+  /**
+   * Add a task to run in parallel
+   */
+  default <T> Future<T> inParallel(RunnableWithException runnable) {
+    return inParallel(runnable, null);
+  }
+
+  /**
+   * Add a task to run in parallel
+   */
+  default <T> Future<T> inParallel(RunnableWithException runnable, T value) {
```

Comment:
What is the use case for passing in a value that is just echoed back via the future?

Reply:
Is it so individual runs can be identified?

Reply:
Maybe not too much. I was copying a similar pattern from the ExecutorService API. I can remove this method until we actually need it.